go/vt/vtadmin/cluster/cluster.go

```diff
@@ -826,9 +831,14 @@ func (c *Cluster) GetCellInfos(ctx context.Context, req *vtadminpb.GetCellInfosR
 		go func(name string) {
 			defer wg.Done()
 
+			if err := c.topoReadPool.Acquire(ctx); err != nil {
+				rec.RecordError(fmt.Errorf("GetCellInfo(%s) failed to acquire topoReadPool: %w", name, err))
+			}
 			resp, err := c.Vtctld.GetCellInfo(ctx, &vtctldatapb.GetCellInfoRequest{
 				Cell: name,
 			})
+			c.topoReadPool.Release()
```

Comment:
no `defer` here to avoid holding on to the topo pool while we're waiting on the mutex